oidc-signature-extension.md

```diff
@@ -262,200 +208,224 @@ The following is a non-normative example of the claims in a Request Object befor
+**Note:** The `https://id.oidc.se/scope/sign` alone does not say anything about the identity of the signing end-user.
+A Relying Party wishing to get this information, which it most likely does, should include additional scopes in the 
+request that declares which identity claims that are wanted. 
 
-| Claim | Description/comment | Requirement |
-| :--- | :--- | :--- |
-| `https://claims.oidc.se/`<br />`1.0/userSignature` | The user signature. For federated signing a user signature MAY be created if the identity provider supports signing. | Optional |
-| `https://claims.oidc.se/`<br />`1.0/signMessageDigest` | A digest of the sign message that was sent by the client in the request. This claim MUST be included iff the client included a sign message in the request and this message was display for, and approved by, the user.  | Optional |
-| `https://claims.oidc.se/1.0/sad` | Signature activation data.<br />*TODO: More will be written about the conditions of including the sad claim.* | Optional |
 
 <a name="relying-party-requirements"></a>
 ## 4. Relying Party Requirements
 
 <a name="requests"></a>
 ### 4.1. Requests
 
-TODO: Refer to iGov, or our own main profile regarding requirements for request parameters.
+A Relying Party wishing to issue a request for signature according to the specification MUST include 
+`https://id.oidc.se/scope/sign` along with the mandatory `openid` as values to the `scope` request parameter. 
 
-A Relying Party wishing to issue a request for signature MUST include either the `https://scopes.oidc.se/1.0/delegatedSign` scope or the `https://scopes.oidc.se/1.0/federatedSign` scope along with the mandatory `openid` scope as values to the `scope` request parameter. It MUST NOT include both signing scopes.
+A request for signature MUST contain the [Signature Request Parameter](#the-signature-request-parameter) and its
+inclusion in the request MUST follow the requirements stated in sections [3.1.1](#placement-of-the-parameter-in-an-authentication-request), [Placement of the Parameter in an Authentication Request](#placement-of-the-parameter-in-an-authentication-request) and [3.1.2](#security-requirements), [Security Requirements](#security-requirements).
 
-A request for signature MUST contain the [Signature Request Parameter Claim](#the-signature-request parameter-claim) and its inclusion in the request MUST follow the requirements stated in sections [3.2.1](#placement-of-the-parameter-in-an-authentication-request), "[Placement of the Parameter in an Authentication Request](#placement-of-the-parameter-in-an-authentication-request)" and [3.2.2](#security-requirements), "[Security Requirements](#security-requirements)".
+The authentication request MUST contain the `prompt` parameter<sup>1</sup> and its value MUST be include both the 
```

Comment:
I'm thinking we might be overcomplicating this from the Relying Party's perspective by making the `prompt` parameter mandatory with two specific values.

Could the specification instead be worded along the lines of "The OP MUST show the userMessage and consent if the authentication request includes a signature request. OPs MUST reject "signature requests" having `prompt=none` (for the reasons already stated in the spec)"

I.e. let the spec give the OP the responsibility of performing the necessary user interactions for a signature operation, rather than requiring the RP to use the `prompt` parameter? Basically - OPs should ignore the `prompt` parameter for signature requests unless `prompt=none` which MUST be rejected.

Just a thought. It's quite possible I'm missing some corner-case best handled by the specific `prompt=login consent`.

Reply:
Yes. I had the same thoughts, but ...  It is not really a good thing to assume implicit values for one parameter (`prompt`) because of the presence of a particular value in another parameter (`scope`). I actually started writing according to your suggestion, but found that it is really better to stick to the OIDC Core definitions as far as possible. And it is not really that cumbersome for a client to specify `prompt=login consent`...

And... A particular OP could choose to be forgiving and treat a missing `prompt` value as `login consent`. But starting to define default behaviour is ... Well, not a good idea.

Reply:
Your reasoning is sound, I agree.